zk-sdk/src/encryption/wasm/elgamal.rs

```diff
@@ -0,0 +1,82 @@
+use {
+    crate::{
+        encryption::{elgamal::ElGamalPubkey, pod::elgamal::PodElGamalPubkey},
+        wasm::display_to_jsvalue,
+    },
+    bytemuck::{Pod, Zeroable},
+    js_sys::{Array, Uint8Array},
+    wasm_bindgen::{prelude::*, JsCast},
+};
+
+#[wasm_bindgen]
+#[derive(Clone, Copy, Default, Pod, Zeroable, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct CompressedElGamalPubkey(PodElGamalPubkey);
+
+#[allow(non_snake_case)]
+#[wasm_bindgen]
+impl CompressedElGamalPubkey {
```

Comment:
Rather than adding the new type, can we just add a separate impl block for `PodElGamalPubkey`? I got this working on my side, do you mind if I push to your branch so you can see what it looks like?

Then we'll also be able to eliminate all of the `wasm` modules!

Reply:
Oh yes, I couldn't quite get the `PodElGamalPubkey` to work in both solana and wasm targets, but if you got it work, then please feel free to push the changes!

Reply:
Ok great, let me know how [0830405](https://github.com/anza-xyz/agave/pull/2996/commits/0830405251c16953dd3b20901ab1692bb380731b) looks!